contracts/strategies/venus/VenusVAIStrategyMainnet.sol

```diff
@@ -8,44 +8,57 @@ import "@pancakeswap/pancake-swap-lib/contracts/token/BEP20/SafeBEP20.sol";
 import "@openzeppelin/contracts-upgradeable/math/MathUpgradeable.sol";
 import "@pancakeswap/pancake-swap-lib/contracts/math/SafeMath.sol";
 
-import "./interface/IVAIVault.sol";
-import "../../base/StrategyBase.sol";
+import "../../base/venus-base/interface/IVAIVault.sol";
+import "../../base/upgradability/BaseUpgradeableStrategy.sol";
 import "../../base/interface/pancakeswap/IPancakeRouter02.sol";
 
-contract VenusVAIStrategy is StrategyBase {
+contract VenusVAIStrategyMainnet is BaseUpgradeableStrategy {
 
   using SafeMath for uint256;
   using SafeBEP20 for IBEP20;
 
-  bool pausedInvesting = false; // When this flag is true, the strategy will not be able to invest. But users should be able to withdraw.
-  IVAIVault public rewardPool = IVAIVault(address(0x0667Eed0a0aAb930af74a3dfeDD263A73994f216));
-  address public wbnb = address(0xbb4CdB9CBd36B01bD1cBaEBF2De08d9173bc095c);
-  address public busd = address(0xe9e7CEA3DedcA5984780Bafc599bD69ADd087D56);
-  address public vai = address(0x4BD17003473389A42DAF6a0a729f6Fdb328BbBd7);
-  address public venus = address(0xcF6BB5389c92Bdda8a3747Ddb454cB7a64626C63);
-
+  IVAIVault public rewardPoolContract;
```

Comment:
Since the strategy is now upgradable, we will likely need to store and preserve `rewardPoolContract` and other variables across upgrades (i.e., having a slot for it like for other variables like vault and underlying).

Reply:
I see, looks like the address is already preserved because it is just a casting of `rewardPool()`. In that case, would be good to move all initializations that would be required to be re-run upon upgrade (that includes `liquidationPath`, for instance) into a separate method, so that it's easy for whoever does the strategy upgrade

Reply:
Alright, these are implications with upgradable contracts that I had not realized before. I will fix it. This does make me unsure that the base strategies `PancakeMasterChefStrategy`, `VenusFoldStrategy` and `VenusWBNBFoldStrategy` are dealing with all this in the right way. Now that I look at them, probably not.